cpp/src/arrow/filesystem/util_internal.h

```diff
@@ -49,6 +49,10 @@ Status NotAFile(const std::string& path);
 ARROW_EXPORT
 Status InvalidDeleteDirContents(const std::string& path);
 
+ARROW_EXPORT
+Result<FileInfoVector> GetGlobFiles(const std::shared_ptr<FileSystem>& filesystem,
+                                    const std::string& glob);
+
```

Comment:
We should add some comments here.  We should point out that all `glob` patterns should be absolute.  For example, I don't think we can do `GetGlobFiles(fs, "*.parquet")`.  This is ok for our purposes (globs are given via `file://` URIs and `file://` URIs must contain absolute paths) but we should make it clear in case someone wants to use this function for another purpose.